### DISCUSSION TITLE
Some confusion about using pysph

### DISCUSSION THREAD

**Discussion by Ordinaryxian** [pypr/pysph#381] [General]
Thanks for your answer! I found that the support radius of the kernel function and the particle search method cannot be modified in my use. Is this modifiable please? It may be possible to write the kernel function by myself, but it is difficult to write the particle search method.
On the other hand, for each time step I need to re-aggregate the particles, calculate their positions, and remove the particles that are not helpful to the system, but this may be difficult to implement in post_step().

**Comment by nauaneed** [comment]
1. I assume you are using the term 'support radius' to refer to the maximum distance $|r|$ such that the kernel $W(r,h)$ is positive, where $h$ is the smoothing length. Now, I assume you'd like to have a kernel, say  a cubic-spline kernel with $r = h$ just like in this [paper](https://doi.org/10.1111/j.1365-2966.2012.21439.x). The usual cubic-spline kernel, that is implemented in PySPH, has a support radius $r = 2h$. If so, you'll need to write the kernel yourself. I am not sure why you'd need to write a particle search method instead of using one already there in PySPH.
2. I am not exactly sure what you mean by 're-aggregate'. But you removing particles is not difficult. You can mark the particles to remove and then remove them within an [Equation](https://pysph.readthedocs.io/en/latest/reference/equations.html#pysph.sph.equation.Equation). Here is an [example](https://github.com/pypr/pysph/blob/16b3b0b70eb3afa29d688efd44d7c01aa0a8b4db/pysph/sph/swe/basic.py#L445).
